### ISSUE TITLE
Ionic application crash

### ISSUE THREAD

**Issue by divanvb** [arnesson/cordova-plugin-firebase#563]
Hi

I've implemented the plugin as per the instructions, changed the strings.xml path in after_prepare.js as well but still as soon as I call the verifyPhoneNumber the app crashes. It worked well for the last 2 days but now it's just not happening. I've also added the SHA of the debug keystore to firebase.

This plugin is very inconsistent, one moment it works without issue and the next it just crashes the application.

Using Ionic with Cordova 7.0.0. 

Any help would be appreciated.

**Comment by robertarnesson** [comment]
do you have a crash log/callstack?

**Comment by blindman** [comment]
I might be experiencing the same issue for `verifyPhoneNumber`.

I can reproduce the issue after I've logged in a bunch of times with the same phone number and physical device. I believe it's casting the exception received incorrectly.

Here is the stack I get:

```
java.lang.ClassCastException: com.google.firebase.FirebaseTooManyRequestsException cannot be cast to com.google.firebase.auth.FirebaseAuthException
                      at org.apache.cordova.firebase.FirebasePlugin$23$1.onVerificationFailed(FirebasePlugin.java:645)
                      at com.google.android.gms.internal.zzdye.zza(Unknown Source:6)
                      at com.google.android.gms.internal.zzdyf.run(Unknown Source:37)
                      at android.os.Handler.handleCallback(Handler.java:790)
                      at android.os.Handler.dispatchMessage(Handler.java:99)
                      at android.os.Looper.loop(Looper.java:164)
                      at android.app.ActivityThread.main(ActivityThread.java:6494)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by zyzmoz** [comment]
I'm having the same issue =/

**Comment by Hirbod** [comment]
Anybody able to fix it? This looks like to happen when calling that stuff without internet connection

**Comment by zyzmoz** [comment]
Just try to reinstall the plugin as in the README.md
`cordova plugin add cordova-plugin-firebase@1.0.0 --save`
Or use this prior version:
`cordova plugin add cordova-plugin-firebase@0.1.25 --save`

Those I shown worked for me !

**Comment by soumak77** [comment]
should be fixed by version 1.0.2

**Comment by Hirbod** [comment]
@soumak77 this isn't fixed in 1.0.2 as there was no change inside "onVerificationFailed" function.

java.lang.ClassCastException: com.google.firebase.FirebaseTooManyRequestsException cannot be cast to com.google.firebase.auth.FirebaseAuthException

cleary describes whats wrong there..
